src/odb_backend.rs

```diff
@@ -0,0 +1,94 @@
+use {raw, Buf, Error, Repository};
+
+use std::ptr;
+
+use util::Binding;
+use libc;
+
+/// A structure to represent a git object database backend.
+pub struct OdbBackendHolder {
```

Comment:
Hm I wonder if we perhaps want to make this a trait or a trait object at some point? It looks like this is similar to the `Transport` trait perhaps in git2-rs?

Reply:
Yes, it is very similar. I attempted a trait implementation, however it quickly became a mess. Is there a way to simultaneously have the structure (backed by the public functions), and the trait to implement custom backends?

Reply:
Yeah doing it as a trait I think will definitely be a relatively significant investment in the sense that it'll take some design. Perhaps that part could be left out from this PR for now?

Reply:
@alexcrichton I'd definitely like to hold off on the trait, however I'd prefer to be able to keep the OdbBackendHolder (or at least a way to use the functionality it provides) as I am using it in a project, but that's entirely up to you.

Reply:
Oh my worry is that the current implementation is not memory safe, so we could just address those sorts of questions in a future PR